docs/managing-secrets-with-vault.md

```diff
@@ -22,38 +22,203 @@ about deploying Vault on Kubernetes for production, see [Vault on Kubernetes
 deployment
 guide](https://developer.hashicorp.com/vault/tutorials/kubernetes/kubernetes-raft-deployment-guide).
 
-_Prerequisites_
+### Prerequisites
 
 * You're logged into the OpenShift Container Platform as a cluster adminstrator.
+* Your OCP has persistent volumes in some way.
 * You've installed `helm` version 3.15 or later.
 
-_Procedure_
-
-* Add the Hashicorp Helm repository:
-  ```bash
-  $ helm repo add hashicorp https://helm.releases.hashicorp.com
-  ```
-* Update all repositories:
-  ```bash
-  $ helm repo update
-  ```
-* Install Hashicorp Vault:
-  * Create a project for Vault:
-    ```bash
-    $ oc new-project vault
-    ```
+### Procedure
+
+_Add the Hashicorp Helm repository_
+```bash
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
+```
+
+_Update all repositories_
+```bash
+$ helm repo update
+```
+
+#### Prepare TLS data
+
+_A few preliminary notes_
+    * In the example bellow, the OCP domain is `ocp.openstack.lab`. You might need to update that value for your own environment.
+    * You can also consume a certificate signed by any other CA. Be sure you set the correct alt_names and CN if you want to use
+      any existing CA.
+
+_Create a custom Certificate Authority_
+```bash
+$ mkdir ~/vault-cert
+$ cd ~/vault-cert
+$ openssl genrsa -out ca.key 4096
+$ openssl req -x509 -new -nodes -key ca.key -sha256 -days 1024 -out ca.crt
+```
+
+_Create a certificate configuration `vault-csr.conf` (set `CN` accordingly)_
+```
+[req]
+default_bits = 4096
+prompt = no
+encrypt_key = yes
+default_md = sha256
+distinguished_name = kubelet_serving
+req_extensions = v3_req
+[kubelet_serving]
+O = system:nodes
+CN = system:node:*.vault.svc.ocp.openstack.lab
+```
+
+_Create an openssl extention file `vault.ext` (set `DNS.2` accordingly)_
+```
+authorityKeyIdentifier=keyid,issuer
+basicConstraints=CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1 = *.vault-internal
+DNS.2 = *.vault-internal.vault.svc.ocp.openstack.lab
+DNS.3 = *.vault
+IP.1 = 127.0.0.1
+```
+
+_Create the vault private key, get a CSR and signed certificate_
+```bash
+$ openssl genrsa -out vault.key 4096
+$ openssl req -new -key vault.key -out vault.csr -config vault-csr.conf
+$ openssl x509 -req -in vault.csr -CA ca.crt -CAkey ca.key \
+    -CAcreateserial -out vault.crt -days 365 -sha256 -extfile vault.ext
+```
+
+_Expose TLS content in OpenShift_
+We now have to let Vault know about the certificate and, if wanted, HA/replicas.
+    * Create a project for Vault:
+      ```bash
+      $ oc new-project vault
+      ```
+    * Create the secret
+      ```bash
+      $ oc -n vault create secret generic vault-ha-tls \
+            --from-file=vault.key=~/vault-cert/vault.key \
+            --from-file=vault.crt=~/vault-cert/vault.crt \
+            --from-file=vault.ca=~/vault-cert/ca.crt
+      ```
```

Comment:
TODO:
- either expose the custom CA to OpenShift so that it trusts it
- or understand why using the kubernetes service issuer isn't working properly for vault to recognize the CA